src/agoradatatools/etl/transform/utils.py

```diff
@@ -0,0 +1,197 @@
+        df[col] = df[col].astype(float)
+
+    obj = {}
+
+    # In order to smooth out the bins and make sure the entire range from 0
+    # to the theoretical maximum value has been found, we create a copy of the
+    # column with both 0 and that maximum value added to it.  We use the copy to calculate
+    # distributions and bins, and subtract the values at the end
+
+    distribution = pd.concat([df[col], pd.Series([0, upper_bound])], ignore_index=True)
+
+    obj["distribution"] = list(
+        pd.cut(
+            distribution, bins=10, precision=3, include_lowest=True, right=True
+        ).value_counts(sort=False)
+    )
+    obj["distribution"][
+        0
+    ] -= 1  # since this was calculated with the artificial 0 value, we subtract it
+    obj["distribution"][
+        -1
+    ] -= 1  # since this was calculated with the artificial upper_bound, we subtract it
+
+    discard, obj["bins"] = list(
+        pd.cut(distribution, bins=10, precision=3, retbins=True)
+    )
+    obj["bins"] = np.around(obj["bins"].tolist()[1:], 2)
+    base = [0, *obj["bins"][:-1]]
+    obj["bins"] = zip(base, obj["bins"])
+    obj["bins"] = list(obj["bins"])
+
+    obj["min"] = np.around(df[col].min(), 4)
+    obj["max"] = np.around(df[col].max(), 4)
+    obj["mean"] = np.around(df[col].mean(), 4)
+    obj["first_quartile"] = np.around(
+        df[col].quantile(q=0.25, interpolation="midpoint")
+    )
+    obj["third_quartile"] = np.around(
+        df[col].quantile(q=0.75, interpolation="midpoint")
+    )
+
+    return obj
+
+
+def count_grouped_total(
+    df: pd.DataFrame, grouping: [str, list], input_colname: str, output_colname: str
+) -> pd.DataFrame:
+    """For each unique item/combination in the column(s) specified by grouping,
+    counts the number of unique items in the column [input_colname] that
+    correspond to that grouping. The calculated counts are put in a new
+    column and named with [output_colname].
+    Args:
+        df (pd.DataFrame): contains columns listed in grouping and
+                           input_colname. May contain other columns as well, but
+                           these will be dropped from the returned data frame.
+        grouping (str or list): a string with a single column name, or a list of
+                                strings for multiple column names
+        input_colname (str): the name of the column to count
+        output_colname (str): the name of the new column with calculated counts
+    Returns:
+        pd.DataFrame: a data frame containing the grouping column(s) and a
+                      new column for output_colname, which contains the count of
+                      unique items in input_colname for each grouping item.
+    """
+    df = (
+        df.groupby(grouping)[input_colname]
+        .nunique()
+        .reset_index()
+        .rename(columns={input_colname: output_colname})
+    )
+    return df
+
+
+def transform_overall_scores(df: pd.DataFrame) -> pd.DataFrame:
```

Comment:
THIS is the function I had a problem with:

I'm on the fence about this function being in `utils.py`: While we do use it in 2-ish different transforms, and therefore it's re-usable, it's **_very_** specific to two data sets and expects those data sets to have specific column names and structure. Whereas the other utility functions here make no assumption about data frame structure. I see a few options:

1. Don't worry about it, leave it in `utils.py`, and make a note in the comments for the function that this is the case. Maybe add some error checking for df structure. Any future Magora transforms would just make sure the dataframe is the same expected format for this function. 
2. Leave it in `utils.py` but abstract some of the structure dependency away from this function, i.e. make some of the hard-coded column names arguments to the function instead, in an attempt to make it at least a little more generic. (this is something I could work on / test if we want to go this route)
3. Have a shared `overall_scores` transform file that includes all scores transforms + this function. I like this option the least...it seems like once we add Magora functions it could be messy.

Reply:
My personal preference is to lean on number 1 then tackle number 2 in a utils.py refactor ticket as I recommended below

Reply:
I think if it is used in more than one transformation it will each be in separated files, it makes sense to keep it in `utils.py`, but if it is only going to be used in one custom transformation file, then it is simpler to have it also be in that separate script.